stan/math/prim/scal/meta/and.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef STAN_MATH_PRIM_SCAL_META_AND_HPP
+#define STAN_MATH_PRIM_SCAL_META_AND_HPP
+
+#include <type_traits>
+
+namespace stan {
+namespace math {
+/**
+ * A helper metaprogram to peform conjuction
+ * on type conditionals for zero or more provided types.
+ * See \link prim/scal/meta/is_var_or_arithmetic.hpp \endlink
+ * for an example.
+ */
+template <typename... T>
+struct and_ : std::true_type {};
```

Comment:
We reserved the trailing underscore convention for member variables, so `and_` needs to be renamed.